### DISCUSSION TITLE
Linux version?

### DISCUSSION THREAD

**Discussion by LepaBrenaa** [blish-hud/Blish-HUD#873] [Blish HUD Core Ideas]
Any possibility to run Blish HUD on Linux?

There is Burrito, but it is a mess, and lacks a lot of functionality.

**Comment by dlamkins** [comment]
The latest releases are capable of running on Linux, but are still opaque due to the WinAPI crimes we're using for transparency not being implemented. This guide below provides a way for those on Linux with certain configurations to now use Blish HUD without native support.

https://gist.github.com/martinlabate/c4e6f08880a009f88dc1edaa4c6cd87a

It does involve some jank, but it's an option for those who really want it.

**Reply by LepaBrenaa** [reply]
Thanks, will take a look at it!

**Reply by Timelightnl** [reply]
I am on Arch Linux and tried this out. I have installed KDE especially for this (I am normally on gnome) but unfortunately I did not get this to work. So I hope someone else has better luck. I don't mind if I have to use a work around to get this working.

**Reply by Sonwon1** [reply]
I had the same problem.

I think there is something wrong with this code you paste into the General tab, in LAUNCH OPTIONS.

"BHDIR=$(realpath "$(find -iname "Blish.HUD*")"); rm -f BlishHUD; ln -s "$BHDIR" BlishHUD; rm -f BlishHUD/BlishHUD.exe; ln -s "$BHDIR/Blish HUD.exe" BlishHUD/BlishHUD.exe; echo -e "@echo off\nstart Gw2-64.exe -provider Steam -autologin\nping -n 60 127.0.0.1>nul\ncd BlishHUD\nstart BlishHUD.exe\nexit" > gw2.bat; DXVK_ASYNC=1 MANGOHUD=0 eval $(echo %command% | sed "s/Guild Wars 2\/Gw2-64\.exe.*$/\"Guild Wars 2\"\/gw2.bat/")"

I think the problem could be in the names used or missing spaces? I haven't had time to dig into yet.

**Comment by LepaBrenaa** [comment]
The necessary steps from the gist above to get it working on Linux aren't satisfying for me.
Plenty of workarounds that don't sit well with me, so I'll have to pass on that.

Is the Linux support being looked at / worked at at all, or is there no intention? Just curious.

**Comment by dlamkins** [comment]
Even native Linux overlays for GW2 (such as Burrito) come with added complexity or workarounds.  So to that end, no, I am not actively pursuing any form of Linux support with my development.  It seems certain that anything we do will still require some amount of band-aiding/hoop-jumping to work.  Among other things, varying desktop managers are a significant barrier to a consistent experience (as Burrito has found).

We, of course, are willing to accept code contributions that make Blish HUD easier to use on Linux or require fewer workarounds as long as those contributions don't impact the experience for our general users.

**Reply by Sonwon1** [reply]
I believe the last update for Burrito fixed the yanky stuff since I have it working using Lutris without any ganks. I do need to test more though so I'll update this if I am wrong.

Edit: last night it was working without problems now today it is not allowing input to Burrito? So I guess it is being yanky.

Edit2: Ok not yanky. It works if you alt tab to the Burrito window then you can click on it and control Burrito. Alt tab back and the pathing shows in GW2 so I would call that a win. Could it work a little better sure, is it worth my time to make it work better while it is fully functional, not for me.

**Comment by Zyian** [comment]
Adding some more information based on some discussion in Discord recently

![image](https://user-images.githubusercontent.com/5796109/230740776-20664cde-f4f7-472c-8169-146984ac6560.png)

**Comment by Taschenbuch** [comment]
![image](https://github.com/blish-hud/Blish-HUD/assets/43114787/ac3a6d48-fa4a-41b1-b1c9-810215be773c)

**Comment by dlamkins** [comment]
Posting my original comment from Discord here so that it is more likely to come up in search results:

Sure thing.  And just to clear some things up.

The primary blocker for proper Linux support is unrelated to us running native or any of our dependencies.  As you've probably seen, we do successfully run on Linux via Wine.  Running natively doesn't directly solve any of the current issues.

The main issue is related to some of the most important aspects of how we function as an overlay.  We use certain WinAPI calls that make the Blish HUD window transparent AND prevent Blish HUD from blocking all keyboard and mouse input to the game.  These calls are not implemented in Wine (I'm not even sure how they could be if we wanted them to due to how they're meant to interact with the window itself which is presumably handled by the window manager which can vary between Linux systems).  Running via wine, this seems to be the only issue we've spotted.

There are other methods for allowing window transparency which do work on Linux (this can be seen with the GW2 launcher window running via Wine), but that do not allow mouse input through the transparent regions.  This would be a new issue that would have to be solved.

Running native doesn't solve either issue.  The WinAPI calls simply wouldn't be available at all on Linux if we run natively and would require platform specific calls to continue to use them for Windows.  Further, there are no equivalent calls on Linux that would universally help us, as far as I am aware, as different DEs / WMs can still cause interruptions.  Burrito doesn't even directly support anything beyond GNOME on x11 for this reason (https://github.com/AsherGlick/Burrito/issues/119).  The behavior just isn't consistent. 

Running native also makes it more difficult for us to communicate with Mumble than when we are running within Wine.  Another problem that would need to be solved.

As overlays are hacky in how they work to begin with, varying DEs and WMs cause for a lot of unintended behaviors.  You can look through some of the other issues reported for Burrito to get an idea of what I mean: https://github.com/AsherGlick/Burrito/issues

Franky, in many ways, we are lucky that Windows just happens to have a handful of APIs that allow us to perform the crimes we need to perform to make this a semi-reasonable experience.

So when folks ask about working on a native version, it's important that all of the other issues are acknowledged and addressed.  Running natively is the easy part.

https://discord.com/channels/531175899588984842/603623649257324545/1094001355074584716

**Reply by Xnyle** [reply]
Would it be possible to at least add an option to the config that lets you choose the background color? I'm not sure, black is the best option when playing with opacity, I'd like to try out white / different grays

**Reply by franzitrone** [reply]
I'm not an expert in the BlishHUD code base, but I did work here and there with the wine source code and I'd be interested in finding a solution on wine's side. @Smoukus could tell me which specific WinAPI functions you're using or point me towards the functions in the BlishHUD code that do this?

**Reply by 256shadesofgrey** [reply]
> I'm not an expert in the BlishHUD code base, but I did work here and there with the wine source code and I'd be interested in finding a solution on wine's side. @Smoukus could tell me which specific WinAPI functions you're using or point me towards the functions in the BlishHUD code that do this?

I think you meant to tag @dlamkins .

**Reply by dlamkins** [reply]
@DeltaTimo https://github.com/blish-hud/Blish-HUD/blob/133f26c2e006bb393d058a8068b783dcf06f30f1/Blish%20HUD/_Utils/WindowUtil.cs#L116-L131

**Comment by FloFri** [comment]
Hello everyone,
I have amazing news. I have actually found a way to have it run near perfect under linux. Will write a more detailed guide later when I have time but here is what I did:

- I did the steps detailed above with the window rules but did set the opacity of the Blish HUD windows to 99%
- Then i used a rewritten version of https://github.com/natask/kwin-effect-smart-invert to inject a fragment shader into the Blish HUD window which sets the alpha to 0 for every pixel which is completely black.

Result:
![image](https://github.com/user-attachments/assets/1daf0b6e-b823-4ca4-97c5-e2b69c829079)

Only super minor graphical glitches (like drop down boxes in the hud are now transparent because they are pure black).

**Reply by Sonwon1** [reply]
Thank you, thank you! I am looking forward to the write up so I can duplicate.

**Comment by FloFri** [comment]
I have now published the plugin for KWin (so this only works if you use a KWin / KDE desktop with KWin Version 5):

https://github.com/FloFri/kwin-blishhud-shader

The steps in short are as follows:

1. Configure your installation as you already did with the "opacity-hack" like in https://gist.github.com/martinlabate/c4e6f08880a009f88dc1edaa4c6cd87a BUT set the opacity for the Blush HUD window (active + inactive) to 99%
2. Download, compile and install the KWin Plugin (https://github.com/FloFri/kwin-blishhud-shader) as described in the Readme.
3. If everything worked you should see the "BlishHUD Shader" plugin under desktop effects in the "Accessibility" section:
![image](https://github.com/user-attachments/assets/eafa86a8-31a4-42d9-848a-d886615d9b47)
4. Deactivate and actvate the plugin once (might not be neccessary)
5. Start Guild Wars 2 and BlishHUD. The HUD should now be transparent.

If someone manages to get it running, I would be glad if that person could write a better howto. Maybe someone is even able to contribute to the plugin.

**Reply by zookee1** [reply]
Thank you for this, @FloFri. 

I can confirm, that it also works on kwin 6.1.5 + plasma 6.1.5 :+1:

**Reply by luimu64** [reply]
> I have now published the plugin for KWin (so this only works if you use a KWin / KDE desktop with KWin Version 5):
> 
> https://github.com/FloFri/kwin-blishhud-shader
> 
> The steps in short are as follows:
> 
>     1. Configure your installation as you already did with the "opacity-hack" like in https://gist.github.com/martinlabate/c4e6f08880a009f88dc1edaa4c6cd87a BUT set the opacity for the Blush HUD window (active + inactive) to 99%
> 
>     2. Download, compile and install the KWin Plugin (https://github.com/FloFri/kwin-blishhud-shader) as described in the Readme.
> 
>     3. If everything worked you should see the "BlishHUD Shader" plugin under desktop effects in the "Accessibility" section:
>        ![image](https://private-user-images.githubusercontent.com/787047/364102734-eafa86a8-31a4-42d9-848a-d886615d9b47.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyODMyMzksIm5iZiI6MTczMDI4MjkzOSwicGF0aCI6Ii83ODcwNDcvMzY0MTAyNzM0LWVhZmE4NmE4LTMxYTQtNDJkOS04NDhhLWQ4ODY2MTVkOWI0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMFQxMDA4NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NTliYWQ0ZWMwODY3MmVkODE4NDFjMDNkNWMyYTBjOWQ5ODFjNjk4ZmUzYmI2NWI0MWE5NWE1N2JhYzkxOGVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.REw6ZravW7ruWsLu_r_Brn2liHp5U57upWqRI-Jds-s)
> 
>     4. Deactivate and actvate the plugin once (might not be neccessary)
> 
>     5. Start Guild Wars 2 and BlishHUD. The HUD should now be transparent.
> 
> 
> If someone manages to get it running, I would be glad if that person could write a better howto. Maybe someone is even able to contribute to the plugin.

I think the FAQ in https://blishhud.com/docs/user/faq#what-platforms-can-blish-hud-run-on should be updated to reflect this new non-compromised way of running under Linux. I was thinking of running this before but thought the way mentioned in the FAQ was the only way so I gave up. Finding this discussion or joining the server to find out is something most people would probably do only after reading the FAQ. I would make pr myself but the only repo for the website I found was archived so not sure where the website source is now.

**Reply by stadsport** [reply]
> I have now published the plugin for KWin (so this only works if you use a KWin / KDE desktop with KWin Version 5):
> 
> https://github.com/FloFri/kwin-blishhud-shader
> 
> The steps in short are as follows:
> 
> 1. Configure your installation as you already did with the "opacity-hack" like in https://gist.github.com/martinlabate/c4e6f08880a009f88dc1edaa4c6cd87a BUT set the opacity for the Blush HUD window (active + inactive) to 99%
> 2. Download, compile and install the KWin Plugin (https://github.com/FloFri/kwin-blishhud-shader) as described in the Readme.
> 3. If everything worked you should see the "BlishHUD Shader" plugin under desktop effects in the "Accessibility" section:
>    ![image](https://private-user-images.githubusercontent.com/787047/364102734-eafa86a8-31a4-42d9-848a-d886615d9b47.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4MzU4NjMsIm5iZiI6MTczMzgzNTU2MywicGF0aCI6Ii83ODcwNDcvMzY0MTAyNzM0LWVhZmE4NmE4LTMxYTQtNDJkOS04NDhhLWQ4ODY2MTVkOWI0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMFQxMjU5MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04OGVhNGRlZTc2NWU3MjdlNGVkNGE1NTJkOTA1MzFmOTRiZTZmNDZjMjdhZTdhZjMxMzY3YjE1ZDA3Njc5M2MyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.1ofiEqNy9Gg8eWJxumA-YtW136Bh1lPgSXakmamR4ew)
> 4. Deactivate and actvate the plugin once (might not be neccessary)
> 5. Start Guild Wars 2 and BlishHUD. The HUD should now be transparent.
> 
> If someone manages to get it running, I would be glad if that person could write a better howto. Maybe someone is even able to contribute to the plugin.

This worked perfect for me. You mentioned in the Readme you may have to disable/enable the Window Effects plugin, and that was true for me too. Disable > Apply > Enable > Apply, after that it's been perfect. 

I wonder if it would be possible to get it added to the KDE Store so it's easy for users to install across different distros in the future? 

Either way, a HUGE thank you for making this!!

**Reply by smooth-baking-powder** [reply]
Hey @FloFri  is there an option to bring it to work on plasma 6 with wayland? (tuxedo 24.04 uses Plasma 6)

Thanks

**Reply by stadsport** [reply]
> Hey @FloFri is there an option to bring it to work on plasma 6 with wayland? (tuxedo 24.04 uses Plasma 6)
> 
> Thanks

I'm not FloFri, but it does work on newer versions too. I have it running on two machines, both running Plasma 6.3 and in Wayland :)

**Reply by smooth-baking-powder** [reply]
> > Hey @FloFri is there an option to bring it to work on plasma 6 with wayland? (tuxedo 24.04 uses Plasma 6)
> > Thanks
> 
> I'm not FloFri, but it does work on newer versions too. I have it running on two machines, both running Plasma 6.3 and in Wayland :)

Hey,

thanks for reply. which dependencies you used?

**Reply by stadsport** [reply]
> > > Hey @FloFri is there an option to bring it to work on plasma 6 with wayland? (tuxedo 24.04 uses Plasma 6)
> > > Thanks
> > 
> > 
> > I'm not FloFri, but it does work on newer versions too. I have it running on two machines, both running Plasma 6.3 and in Wayland :)
> 
> Hey,
> 
> thanks for reply. which dependencies you used?

Check out the main project page here:
https://github.com/FloFri/kwin-blishhud-shader

In the installation instructions, they show which dependences are needed based on your distro. Since Tuxedo is based on Ubuntu, you should be able to run their command to install the necessary dependences on Debian/Ubuntu:

```sudo apt install git cmake g++ gettext extra-cmake-modules qttools5-dev libqt5x11extras5-dev libkf5configwidgets-dev libkf5crash-dev libkf5globalaccel-dev libkf5kio-dev libkf5notifications-dev kinit-dev kwin-dev```

(Hopefully that formatting looks correct, I'm on mobile) 

I'm running Fedora so slightly different command for me, but the Debian command should work for you. Otherwise, try the ```cmake``` command at the very end of their instructions, and if any dependencies come up missing, try installing them manually with apt.

**Reply by smooth-baking-powder** [reply]
> In the installation instructions, they show which dependences are needed based on your distro. Since Tuxedo is based on Ubuntu, you should be able to run their command to install the necessary dependences on Debian/Ubuntu:
> 
> `sudo apt install git cmake g++ gettext extra-cmake-modules qttools5-dev libqt5x11extras5-dev libkf5configwidgets-dev libkf5crash-dev libkf5globalaccel-dev libkf5kio-dev libkf5notifications-dev kinit-dev kwin-dev`
> 
> (Hopefully that formatting looks correct, I'm on mobile)
> 
> I'm running Fedora so slightly different command for me, but the Debian command should work for you. Otherwise, try the `cmake` command at the very end of their instructions, and if any dependencies come up missing, try installing them manually with apt.

Hey,

i tried, but i still get this error:
```
/Appdata/git/kwin-blishhud-shader/build$ sudo apt install git cmake g++ gettext extra-cmake-modules qttools5-dev libqt5x11extras5-dev libkf5configwidgets-dev libkf5crash-dev libkf5globalaccel-dev libkf5kio-dev libkf5notifications-dev kinit-dev kwin-dev
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
git ist schon die neueste Version (1:2.43.0-1ubuntu7.2).
cmake ist schon die neueste Version (3.30.5-0zneon+24.04+noble+release+build3).
g++ ist schon die neueste Version (4:13.2.0-7ubuntu1).
gettext ist schon die neueste Version (0.21-14ubuntu2).
extra-cmake-modules ist schon die neueste Version (6.10.0-0zneon+24.04+noble+release+build9).
qttools5-dev ist schon die neueste Version (5.15.16+p24.04+vrelease+git20241207.0235-0).
libqt5x11extras5-dev ist schon die neueste Version (5.15.16+p24.04+vrelease+git20241207.0420-0).
libkf5configwidgets-dev ist schon die neueste Version (5.116.0-1zneon+24.04+noble+release+build3).
libkf5crash-dev ist schon die neueste Version (5.116.0-1zneon+24.04+noble+release+build3).
libkf5globalaccel-dev ist schon die neueste Version (5.116.0-1zneon+24.04+noble+release+build3).
libkf5kio-dev ist schon die neueste Version (5.116.0-1zneon+24.04+noble+release+build3).
libkf5notifications-dev ist schon die neueste Version (5.116.0-1zneon+24.04+noble+release+build4).
kinit-dev ist schon die neueste Version (5.116.0-1zneon+24.04+noble+release+build3).
kwin-dev ist schon die neueste Version (4:6.2.5-0zneon+24.04+noble+release+build23).
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.
backpulver@Backpulver-I:~/Appdata/git/kwin-blishhud-shader/build$ cmake ..
-- Could NOT find KF6ConfigWidgets: found neither KF6ConfigWidgetsConfig.cmake nor kf6configwidgets-config.cmake 
-- Could NOT find KF6ConfigWidgets (missing: KF6ConfigWidgets_DIR)
-- Could NOT find KF6ConfigWidgets: found neither KF6ConfigWidgetsConfig.cmake nor kf6configwidgets-config.cmake 
CMake Error at /usr/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find KF6 (missing: ConfigWidgets) (Required is at least version
  "6")

      Reason given by package: Missing the following required components: ConfigWidgets

Call Stack (most recent call first):
  /usr/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/ECM/find-modules/FindKF6.cmake:98 (find_package_handle_standard_args)
  CMakeLists.txt:42 (find_package)


-- Configuring incomplete, errors occurred!
backpulver@Backpulver-I:~/Appdata/git/kwin-bli
```

**Reply by stadsport** [reply]
> > In the installation instructions, they show which dependences are needed based on your distro. Since Tuxedo is based on Ubuntu, you should be able to run their command to install the necessary dependences on Debian/Ubuntu:
> > `sudo apt install git cmake g++ gettext extra-cmake-modules qttools5-dev libqt5x11extras5-dev libkf5configwidgets-dev libkf5crash-dev libkf5globalaccel-dev libkf5kio-dev libkf5notifications-dev kinit-dev kwin-dev`
> > (Hopefully that formatting looks correct, I'm on mobile)
> > I'm running Fedora so slightly different command for me, but the Debian command should work for you. Otherwise, try the `cmake` command at the very end of their instructions, and if any dependencies come up missing, try installing them manually with apt.
> 
> Hey,
> 
> i tried, but i still get this error:
> 
> ```
..........
> ```
The error appears to be because some KDE Frameworks 6 (KF6) packages are not installed, though oddly enough looking at the dependencies command for Debian, they look to be calling for KF5 packages, while the Fedora script (which I ran) calls for KF6.

So if those errors are to be believed, try installing the KF6 equivalent packages:
```sudo apt install libkf6configwidgets-dev libkf6crash-dev libkf6globalaccel-dev libkf6kio-dev libkf6notifications-dev kwin6-dev```

And just to be safe you might need these:
```sudo apt install qt6-base-dev qt6-tools-dev libkf6coreaddons-dev```

Maybe give those a shot and try it again

**Reply by smooth-baking-powder** [reply]
> Maybe give those a shot and try it again

kwin6-dev is not known package :/

Still error after installed all other:
```
/Appdata/git/kwin-blishhud-shader/build$ cmake ..
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so
-- Found KF6ConfigWidgets: /usr/lib/x86_64-linux-gnu/cmake/KF6ConfigWidgets/KF6ConfigWidgetsConfig.cmake (found version "6.11.0")
-- Found KF6: success (found suitable version "6.11.0", minimum required is "6") found components: ConfigWidgets
-- Could NOT find KF6KCMUtils (missing: KF6KCMUtils_DIR)
-- Could NOT find KF6KCMUtils: found neither KF6KCMUtilsConfig.cmake nor kf6kcmutils-config.cmake 
CMake Error at /usr/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find KF6 (missing: KCMUtils) (Required is at least version "6")

      Reason given by package: Missing the following required components: KCMUtils

Call Stack (most recent call first):
  /usr/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/ECM/find-modules/FindKF6.cmake:98 (find_package_handle_standard_args)
  CMakeLists.txt:45 (find_package)


-- Configuring incomplete, errors occurred!
```

**Reply by Technetium1** [reply]
@smooth-baking-powder it has been at v6 for quite some time now, try to install [kf6-kcmutils](https://launchpad.net/ubuntu/+source/kf6-kcmutils)

**Reply by smooth-baking-powder** [reply]
> @smooth-baking-powder it has been at v6 for quite some time now, try to install [kf6-kcmutils](https://launchpad.net/ubuntu/+source/kf6-kcmutils)

its already installed.
```
sudo apt-get install kf6-kcmutils
[sudo] Passwort für backpulver: 
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
kf6-kcmutils ist schon die neueste Version (6.11.0-0zneon+24.04+noble+release+build14).
kf6-kcmutils wurde als manuell installiert festgelegt.
Die folgenden Pakete wurden automatisch installiert und werden nicht mehr benötigt:
  libdisplay-info1 libgps30t64 libkdecorations2-6 libkdecorations2private11 libpam-kwallet6 plasma-discover-common
Verwenden Sie »sudo apt autoremove«, um sie zu entfernen.
0 aktualisiert, 0 neu installiert, 0 zu entfernen und 0 nicht aktualisiert.
```

**Reply by smooth-baking-powder** [reply]
I got it with 

**sudo apt-get install kf6-kcmutils-dev**


Log:
```
Paketlisten werden gelesen… Fertig
Abhängigkeitsbaum wird aufgebaut… Fertig
Statusinformationen werden eingelesen… Fertig
Die folgenden Pakete wurden automatisch installiert und werden nicht mehr benötigt:
  libdisplay-info1 libgps30t64 libkdecorations2-6 libkdecorations2private11 libpam-kwallet6 plasma-discover-common
Verwenden Sie »sudo apt autoremove«, um sie zu entfernen.
Die folgenden NEUEN Pakete werden installiert:
  kf6-kcmutils-dev
0 aktualisiert, 1 neu installiert, 0 zu entfernen und 1 nicht aktualisiert.
Es müssen 159 kB an Archiven heruntergeladen werden.
Nach dieser Operation werden 367 kB Plattenplatz zusätzlich benutzt.
Holen:1 https://plasma6.tuxedocomputers.com noble/main amd64 kf6-kcmutils-dev amd64 6.11.0-0zneon+24.04+noble+release+build14 [159 kB]
Es wurden 159 kB in 0 s geholt (608 kB/s).
Vormals nicht ausgewähltes Paket kf6-kcmutils-dev wird gewählt.
(Lese Datenbank ... 396469 Dateien und Verzeichnisse sind derzeit installiert.)
Vorbereitung zum Entpacken von .../kf6-kcmutils-dev_6.11.0-0zneon+24.04+noble+release+build14_amd64.deb ...
Entpacken von kf6-kcmutils-dev (6.11.0-0zneon+24.04+noble+release+build14) ...
kf6-kcmutils-dev (6.11.0-0zneon+24.04+noble+release+build14) wird eingerichtet ...
backpulver@Backpulver-I:~/Appdata/git/kwin-blishhud-shader/build$ cmake ..
-- Found KF6KCMUtils: /usr/lib/x86_64-linux-gnu/cmake/KF6KCMUtils/KF6KCMUtilsConfig.cmake (found version "6.11.0")
-- Found KF6: success (found suitable version "6.11.0", minimum required is "6") found components: KCMUtils
-- Found XKB: /usr/lib/x86_64-linux-gnu/libxkbcommon.so (found suitable version "1.6.0", minimum required is "0.5.0")
-- Found X11: /usr/include
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so
-- Looking for XOpenDisplay in /usr/lib/x86_64-linux-gnu/libX11.so;/usr/lib/x86_64-linux-gnu/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found Wayland_Server: /usr/lib/x86_64-linux-gnu/libwayland-server.so (found version "1.23.0")
-- Found Wayland: /usr/lib/x86_64-linux-gnu/libwayland-server.so (found version "1.23.0") found components: Server
-- Found epoxy: /usr/lib/x86_64-linux-gnu/libepoxy.so (found version "1.5.10")
-- Found KWin Version: 6.3.2
-- Found KWinEffect API Version: 236
-- Found XCB_XCB: /usr/lib/x86_64-linux-gnu/libxcb.so (found version "1.15")
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so (found version "1.15") found components: XCB
-- The following OPTIONAL packages have been found:

 * OpenGL
 * Qt6QuickTools (required version >= 6.8.2)
 * Qt6DBusTools (required version >= 6.8.2)
 * Qt6QmlTools (required version >= 6.8.2)
 * XKB (required version >= 0.5.0), XKB API common to servers and clients., <http://xkbcommon.org>
 * WrapVulkanHeaders
 * Qt6CoreTools (required version >= 6.8.2)
 * Qt6GuiTools (required version >= 6.8.2)

-- The following REQUIRED packages have been found:

 * ECM (required version >= 5.100)
 * KF6ConfigWidgets (required version >= 6.11.0)
 * KF6KCMUtils (required version >= 6)
 * KF6 (required version >= 6)
 * KWin
 * epoxy
 * XCB, X protocol C-language Binding, <https://xcb.freedesktop.org/>
 * KWinDBusInterface

-- Configuring done (0.9s)
-- Generating done (0.0s)
-- Build files have been written to: /home/backpulver/Appdata/git/kwin-blishhud-shader/build

```

**Comment by dlamkins** [comment]
FYI, I've added a channel to our Blish HUD Discord where this kind of stuff can also be discussed:

https://discord.com/channels/531175899588984842/1290012018677645352

**Comment by Xnyle** [comment]
Certainly great work youre doing with those KWin "hacks" but the thing is, it is only running on KWin. So for me thats more or less as helpful as a solution that would only run on Win without the K.

Sadly BlissHUD  stopped working on latest wine anyway, some mono/eglib NPE assertion. So I was fed up enough and got Burrito working again. More or less.

BlissHUD also is moving to a more recent .NET framework which will probably break linux compatibility even more.
https://github.com/blish-hud/Blish-HUD/pull/924

Actually I don't understand why those addons use some obscure Mono/Windows only or Godot solution at all for the UI? 
When you could just use Electron and then program the UI in Java/Typescript.

It is platform independent and would also attract way more developers, cause Javascipt/Typescript is easy and everywhere.

Electron natively supports transparent windows and also has setIgnoreMouseEvents / blur which at first glance seems to be exactly what is needed.

Actually.., For the Pathing Module I wonder how hard can it be to just connect to burrito link, read in a taco XML file, steal some 3D transform functions from Burritos Ruby Scripts and then draw some vectors on a canvas element...

**Reply by smooth-baking-powder** [reply]
Any updates here? i cant bring blishhud to work

**Comment by Absolemo** [comment]
https://www.youtube.com/watch?v=jKdek4sIf84
This method also works smoothly within Steam.

**Comment by namoninja** [comment]
https://gitlab.com/namoninja/proton-nn

Native WINE patch

**Reply by Technetium1** [reply]
Looking forward to seeing this completed and upstreamed!

**Comment by Technetium1** [comment]
I have been using TaimiHUD for some months now, and it is nearing feature parity in my opinion. Especially with API integration being almost complete.

https://github.com/TaimiHUD/TaimiHUD can be installed via https://github.com/RaidcoreGG/Nexus/

**Reply by Sonwon1** [reply]
I've been using TamiHUD for about one month and have had zero problems. No hoops to jump through just load and play.
